odoo/orm/domains.py

```diff
@@ -1291,16 +1291,18 @@ def _value_to_datetime(value):
             warnings.warn("Use naive datetimes in domains")
             value = value.astimezone(timezone.utc).replace(tzinfo=None)
         return value, False
-    if isinstance(value, SQL) or value is False:
-        return value, False
+    if value is False:
+        return False, True
     if isinstance(value, str):
         dt, _ = _value_to_datetime(datetime.fromisoformat(value))
         return dt, len(value) == 10
     if isinstance(value, date):
         return datetime.combine(value, time.min), True
     if isinstance(value, COLLECTION_TYPES):
         value, is_day = zip(*(_value_to_datetime(v) for v in value))
-        return OrderedSet(value), any(is_day)
+        return OrderedSet(value), all(is_day)
```

Comment:
I think we can now define the second return value as real `is_date` and typing the converter.
when `value is False`, return `False, False`
when `isinstance(value, COLLECTION_TYPES)`, return `OrderedSet(value), False`
`is_date` means the first return value is **one** `datetime` for `date`

Reply:
No, it means that we have no datetimes.

Reply:
It means `no datetimes` now. But its variable is named `is_day` while `(not is_day) != no_datetimes`
I feel it the logic is quite tricky to use especially when `value` is `COLLECTION_TYPES`
Since we don't use `is_day` in the for `'in'` and `'not in'`, I propose we change the sementic of the second value.
Unless maybe the `no_datetime` logic is useful in the future when we want to hanle equality for `date`?

Reply:
Broke 2 tests, I think that we should keep the logic as it is in the PR.

Reply:
OK. Thanks. We can keep the current code.
But since the meaning of the second returned value is already changed.
Could you add a doc to the method?
Then we can merge this 👍